Kernel/Graphics/VirtIOGPU/GPU3DDevice.cpp

```diff
@@ -122,10 +121,10 @@ ErrorOr<void> VirtIOGPU3DDevice::ioctl(OpenFileDescription& description, unsigne
             .padding = 0,
         };
         SpinlockLocker locker(m_graphics_adapter->operation_lock());
-        auto resource_id = m_graphics_adapter->create_3d_resource(resource_spec).value();
-        m_graphics_adapter->attach_resource_to_context(resource_id, per_context_state->context_id());
-        m_graphics_adapter->ensure_backing_storage(resource_id, per_context_state->transfer_buffer_region(), 0, NUM_TRANSFER_REGION_PAGES * PAGE_SIZE);
-        spec.created_resource_id = resource_id;
+        auto resource_id = TRY(m_graphics_adapter->create_3d_resource(resource_spec));
```

Comment:
Technically we should be freeing the resource if any of these fail now (we didn't need to do this before because up until now we'd crash otherwise), but for now it is probably sufficient to add a FIXME to do so.